src/app/pages/courses/courses.page.spec.ts

```diff
@@ -30,11 +50,34 @@ describe('CoursesComponent', () => {
     expect(getByText(exampleCourse.title)).toBeTruthy();
   });
 
-  it('should call the delete method from courses after clicking delete button of the course', () => {
+  it('should show delete modal after clicking delete button', () => {
+    const { getByText, click } = component;
+    const dialog = TestBed.inject(MatDialog);
+    spyOn(dialog, 'open');
+    click(getByText('Delete'));
+    expect(dialog.open).toHaveBeenCalled();
+  });
+
+  it('should remove course after click "confirm" on modal', () => {
+    const { getByText, click } = component;
+    const dialog = TestBed.inject(MatDialog);
+    const service = TestBed.inject(CoursesService);
+    // QUESTION: IS IT GOOD TEST SCENARIO AND IMPLEMENTATION OF TEST?
+    spyOn(dialog, 'open').and.returnValue(dialogRefFactory(true) as any);
+    spyOn(service, 'removeItem');
+    click(getByText('Delete'));
+    expect(service.removeItem).toHaveBeenCalledWith(exampleCourse);
+  });
+
+  it('should not remove course after click "cancel" on modal', () => {
     const { getByText, click } = component;
-    spyOn(instance, 'onCardDelete');
+    const dialog = TestBed.inject(MatDialog);
+    const service = TestBed.inject(CoursesService);
+    // QUESTION: IS IT GOOD TEST SCENARIO AND IMPLEMENTATION OF TEST?
```

Comment:
Well... rather not. 

I would suggest having a dialog wrapper service and a completely abstract MatDialog. I think that we have a similar approach somewhere in EFX projects. With this approach you completely ignore that you have a dialog, you're only concerned with what service returns.